File: src/cpp/example_players/legalplayer.cpp
#include "legalplayer.h"

using namespace K273;
using namespace GGPLib;
using namespace LegalPlayer;

int Player::onNextMove(double end_time) {
    LegalState* ls = this->sm->getLegalState(this->our_role_index);
    return ls->getLegal(0);
}

File: src/ggplib/player/legal_player.py
from ggplib.player.base import MatchPlayer


class LegalPlayer(MatchPlayer):
    ' plays randomly '

    def on_next_move(self, finish_time):
        # get the statemachine...
        sm = self.match.sm

        # constraint: state machine will be correct
        ls = sm.get_legal_state(self.match.our_role_index)

        # a first choice
        return ls.get_legal(0)
